### PR TITLE
Possible improvements for library

### PR DESCRIPTION
**Pull request by tanepiper**
This PR adds:

* Add types for Observable Notification and Network
* Add documentation to operators
* Change network.connection API to use fromEventPattern
* Add errors where approriate

These probably need some test coverage running in a real browser such as Karma tests.

### CODE REVIEW

File: src/web-api/matchMedia.ts [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
+ * A RxJS operator for getting the results of the
+ * {@link https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList|Media Query API}
+ * @param mediaQueryList The media query list to observe
+ * @returns An Observable of events from the media query
+ */
 function fromMediaListQuery(mediaQueryList: MediaQueryList): Observable<MediaQueryListEvent> {
   return fromEventPattern(
-    handler => mediaQueryList.addListener(handler),
-    handler => mediaQueryList.removeListener(handler)
+    handler => mediaQueryList.addEventListener('change', handler),
```
Comment (niklas-wortmann) [inline src/web-api/matchMedia.ts:13]: how do you know about this 'change' parameter, I didn't found anything about that in the documentation?

File: src/observer/resizeObserver.ts [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
 export function fromResizeObserver(
   target: Element,
   options?: ResizeObserverOptions
 ): Observable<ResizeNotification> {
   return new Observable(subscriber => {
-    const resizeObserver = new ResizeObserver((entries, observer) => {
-      subscriber.next({ entries, observer });
-    });
+    const resizeObserver = new ResizeObserver(
+      (entries: readonly ResizeObserverEntry[], observer: ResizeObserver) => {
```
Comment (niklas-wortmann) [inline src/observer/resizeObserver.ts]: IMO these typings are unnecessary and kinda bloat the code, could you remove these and also from the other observer operators

**Review by niklas-wortmann — CHANGES_REQUESTED** [review]
Many thanks for this pull requests. It's amazing, I have just requested some minor changes

File: src/web-api/matchMedia.ts (hunk 1 above)
Comment (tanepiper) [reply src/web-api/matchMedia.ts:13]: in `lib.dom.d.ts` is the following:

image

**Comment by niklas-wortmann** [comment]
This is amazing! thanks a lot for all the work you put into that! Are you still working on something or can I merge this?

**Comment by tanepiper** [comment]
@JWO719 if the tests pass green it's ready to merge :)

At the moment they are highly mocked.  I tried investigating Jest in the browser, I can get puppeteer to work but not webdriver. Also JSDom causes similar issues. 

Without figuring out a way to test them natively (and not sure geolocation is possible) this was the best solution I could come up with.

**Comment by tanepiper** [comment]
@JWO719 See #2 for more improvements on this branch.  That change has a bigger change in the repo itself (using @nrwl/nx) but now also has the outline of an app that can be used to run in browser, also run E2E tests (as Jest will only ever be mockable).

I've left this branch open to give the choice of accepting either 📦

**Comment by niklas-wortmann** [comment]
I'll close this in favor of #2

**Pull request closed without merging**